features/fields.feature

```diff
@@ -0,0 +1,341 @@
+      | status   | 400                                                                                                         |
+      | detail   | De foutieve parameter(s) zijn: fields.                                                                      |
+      | code     | paramsValidation                                                                                            |
+      | instance | /haalcentraal/api/brp/personen                                                                              |
+      En heeft het object de volgende 'invalidParams' gegevens
+      | code   | name   | reason                                           |
+      | fields | fields | Deel van de parameterwaarde niet correct: <pad>. |
+
+      Voorbeelden:
+      | pad                                  |
+      | voorletters                          |
+      | voornamen                            |
+      | adellijkeTitelPredicaat              |
+      | adellijkeTitelPredicaat.code         |
+      | adellijkeTitelPredicaat.omschrijving |
+      | adellijkeTitelPredicaat.soort        |
+      | voorvoegsel                          |
+      | geslachtsnaam                        |
+      | datum                                |
+      | plaats                               |
+      | land                                 |
+      | type                                 |
+      | code                                 |
+      | plaats.code                          |
+      | land.code                            |
+      | omschrijving                         |
+      | plaats.omschrijving                  |
+      | land.omschrijving                    |
+      | soort                                |
+      | datumIngangGeldigheid                |
+      | indicatieOnbekend                    |      
+
+  Rule: Wanneer velden van polymorf gegevensgroep wordt gevraagd, wordt altijd het 'type' veld van de gegevensgroep terug gegeven
+    - wanneer het gegeven geen waarde heeft (ook niet onbekend) wordt de gegevensgroep niet geleverd en dus ook 'type' niet teruggegeven
+
+    Abstract Scenario: Gevraagd veld is een polymorf gegevensgroep van de persoon
+      Gegeven de persoon heeft de volgende 'geboorte' gegevens
+      | naam  | waarde      |
+      | datum | <GBA datum> |
+      Als personen wordt gezocht met de volgende parameters
+      | naam                | waarde                             |
+      | type                | RaadpleegMetBurgerservicenummer    |
+      | burgerservicenummer | 999995078                          |
+      | fields              | burgerservicenummer,geboorte.datum |
+      Dan heeft de persoon met burgerservicenummer '999995078' alleen de volgende gegevens
+      | naam                | waarde    |
+      | burgerservicenummer | 999995078 |
+      En heeft de persoon met burgerservicenummer '999995078' alleen de volgende 'geboorte' gegevens

```

Comment:
Natuurlijk kan iedereen a.d.h.v. de openapi.yaml specificatie zelf bepalen in welke situatie welke velden terugverwacht mogen worden. Ik vraag me alleen af of men a.d.h.v de volgende tabel niet verwacht dat bijv. bij het type 'Datum' ook het veld 'datum.onbekend' in de response zit.

Is de volgende vorm misschien een optie?

```
| polymorf type  | veld        | waarde         |
| Datum          | datum.type  | Datum          |
| Datum          | datum.datum | <datum>        |
| JaarMaandDatum | datum.type  | JaarMaandDatum |
| JaarMaandDatum | datum.jaar  | <jaar>         |
| JaarMaandDatum | datum.maand | <maand>        | 
...
```

Reply:
volgens de tabel die er nu zit, zal bij 'Datum' veld 'onbekend' leeg zijn. je bedoelt dat men dan verwacht dat dit veld 'onbekend' dan in de response zit als 'onbekend': null o.i.d.? Dat is niet zo erg denk ik, als maar duidelijk is dat dit veld dan geen waarde heeft.

Het is nu op deze manier opgenomen omdat dan bij heel veel scenario's in vrijwel alle features op exact dezelfde manier wordt aangegeven wat wordt verwacht. Dat maakt ook het automatiseren tot uitvoerbare test veel eenvoudiger.